cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -1027,6 +1031,24 @@ protected void jvmPerformanceOptions(List<EnvVar> envVars) {
      * @throws IOException
      */
     protected Map<String, CertAndKey> maybeCopyOrGenerateCerts(CertManager certManager, Secret secret, int replicasInSecret, CertAndKey caCert, BiFunction<String, Integer, String> podName) throws IOException {
+        return maybeCopyOrGenerateCerts(certManager, secret, replicasInSecret, caCert, podName, null, Collections.EMPTY_MAP);
+    }
+
+    /**
+     * Copy already existing certificates from provided Secret based on number of effective replicas
+     * and maybe generate new ones for new replicas (i.e. scale-up)
+     *
+     * @param certManager CertManager instance for handling certificates creation
+     * @param secret The Secret from which getting already existing certificates
+     * @param replicasInSecret How many certificates are in the Secret
+     * @param caCert CA certificate to use for signing new certificates
+     * @param podName A function for resolving the Pod name
+     * @param externalBootstrapAddress External address to the bootstrap service
+     * @param externalAddresses Map with external addresses under which the individual pods are available
+     * @return Collection with certificates
+     * @throws IOException
+     */
+    protected Map<String, CertAndKey> maybeCopyOrGenerateCerts(CertManager certManager, Secret secret, int replicasInSecret, CertAndKey caCert, BiFunction<String, Integer, String> podName, String externalBootstrapAddress, Map<String, String> externalAddresses) throws IOException {
```

Comment:
It seems `externalAddresses` is populated by this method, so it's effectively a result of the method rather than a parameter. I think it would be much cleaner to write a result class to hold the `Map<String, CertAndKey>` and `Map<String, String> externalAddresses`, rather than passing a Map to be populated by the method.

Reply:
No, the `externalAddresses` is populated by `kafkaReplicaRoutesReady` in `KafkaAssemblyOperator`. `maybeCopyOrGenerateCerts` is only reading the `externalAddresses` to use them as subjects in the certificates.